android/src/main/kotlin/com/simform/audio_waveforms/AudioRecorder.kt

```diff
@@ -142,7 +144,8 @@ class AudioRecorder : PluginRegistry.RequestPermissionsResultListener {
         return result == PackageManager.PERMISSION_GRANTED
     }
 
-    fun checkPermission(result: MethodChannel.Result, activity: Activity?) {
+    fun checkPermission(result: MethodChannel.Result, activity: Activity?, successCallback: RequestPermissionsSuccessCallback) {
+        this.successCallback = successCallback
```

Comment:
No @ujas-m-simformsolutions it will call no matter either granted the permission or decline the permission. it will return the result based on user action.